airbyte-webapp/src/components/EntityTable/components/StatusCell.tsx

```diff
@@ -48,21 +57,24 @@ const StatusCell: React.FC<IProps> = ({ enabled, isManual, id, isSyncing, onSync
         <Switch checked={enabled} onChange={onSwitchChange} disabled={!allowSync} loading={isLoading} />
       </div>
     );
-  }
-
-  if (isSyncing) {
-    return (
+  } else if (isSyncing) {
+    ControlComponent = (
       <ProgressMessage>
         <FormattedMessage id="tables.progress" />
       </ProgressMessage>
     );
+  } else {
+    ControlComponent = (
```

Comment:
Rather than re-assign ControlComponent, I think it would be good to turn it into its own component and conditionally return the right JSX in there.

Reply:
the initial idea was the same, but @edmundito suggest to keep everything in one component to make merging with https://github.com/airbytehq/airbyte/pull/19226 easier. after this PR will be merged, I am planing to return to my initial idea with separate component

Reply:
Oh, yeah, sure. We also need to take into account that. 👍

Reply:
fixed this, since @edmundito was merged